src/wallet/wallet2.cpp

```diff
@@ -13046,6 +13055,77 @@ void wallet2::throw_on_rpc_response_error(const boost::optional<std::string> &st
   THROW_WALLET_EXCEPTION_IF(*status != CORE_RPC_STATUS_OK, tools::error::wallet_generic_rpc_error, method, m_trusted_daemon ? *status : "daemon error");
 }
 //----------------------------------------------------------------------------------------------------
+
+bool wallet2::save_to_file(const std::string& path_to_file, const std::string& raw, bool is_hex) const
+{
+  if (is_hex || m_export_format == ExportFormat::Raw)
+  {
+    return epee::file_io_utils::save_string_to_file(path_to_file, raw);
+  }
+
+  std::string h = HEX_OUTPUT_MAGIC + "\n";
+  h.append(epee::string_tools::buff_to_hex_nodelimer(raw));
+
+  // Split the long string into multiple lines, similarly to how openssl
+  // prints certificates.
+  for (size_t i = 63; i < h.length(); i += 64)
+  {
+    h.insert(i, 1, '\n');
```

Comment:
It should be faster to allocate an entirely new string (with `reserve`/`resize` in one shot), then copy the portions into it. This technique is constantly doing `memmove`'s with the _same_ bytes (i.e. the last bytes of the string are being moved N times), and I don't see an easy way for the compiler to optimize that away.

The other technique is to resize the existing string, and do the `memmove` from-the-back-first manually while adding the newlines.

This will definitely matter for large files, but I'm not sure how large the files will get for the use cases.

Reply:
All right, reimplemented. Not the most readable piece of code but it should get the job done :) .